gattc_sd.go

```diff
@@ -137,19 +157,19 @@ func (s *DeviceService) DiscoverCharacteristics(uuids []UUID) ([]DeviceCharacter
 		return nil, errAlreadyDiscovering
 	}
 
-	// Make a list of UUIDs in SoftDevice short form, for easier comparing.
-	shortUUIDs := make([]C.ble_uuid_t, len(uuids))
-	for i, uuid := range uuids {
+	// Make a map of UUIDs in SoftDevice short form, for easier comparing.
+	shortUUIDs := make(map[uuidWrapper]C.ble_uuid_t)
+	for _, uuid := range uuids {
 		var errCode uint32
-		shortUUIDs[i], errCode = uuid.shortUUID()
+		shortUUIDs[uuid], errCode = uuid.shortUUID()
 		if errCode != 0 {
 			return nil, Error(errCode)
 		}
 	}
 
 	// Request characteristics one by one, until all are found.
 	numFound := 0
-	characteristics := make([]DeviceCharacteristic, len(uuids))
+	characteristics := []DeviceCharacteristic{}
```

Comment:
The same logic regarding heap allocations of slices applies here.

Reply:
Already handled.